lib/cocoapods/installer/analyzer.rb

```diff
@@ -522,12 +500,12 @@ def generate_pod_targets(specs_by_target)
           specs_by_target.flat_map do |target_definition, specs|
             grouped_specs = specs.group_by(&:root).values.uniq
             pod_targets = grouped_specs.flat_map do |pod_specs|
-              generate_pod_target([target_definition], pod_specs).scoped(dedupe_cache)
+              generate_pod_target([target_definition], pod_specs.map(&:spec), pod_specs.all?(&:used_by_tests_only?)).scoped(dedupe_cache)
             end
 
             pod_targets.each do |target|
-              dependencies = transitive_dependencies_for_specs(target.specs, target.platform, specs).group_by(&:root)
-              test_dependencies = transitive_dependencies_for_specs(target.specs.select(&:test_specification?), target.platform, all_specs).group_by(&:root)
```

Comment:
When `deduplicate_targets` is OFF this logic was completely broken. `all_specs` is not even defined here.

Goes to show how many people actually turn `deduplicate_targets` OFF